compiler/src/dotty/tools/dotc/reporting/MessageKind.scala

```diff
@@ -0,0 +1,25 @@
+package dotty.tools.dotc.reporting
+
+/** The various message kinds that can be used inside of a Message
+ *  @param name What will be pontentially displayed to the user
+ */
+enum MessageKind(name: String):
+  case NoKind extends MessageKind("No Kind")
+  case Syntax extends MessageKind("Syntax")
+  case Type extends MessageKind("Type")
+  case TypeMismatch extends MessageKind("Type Mismatch")
+  case Naming extends MessageKind("Naming")
+  case Declaration extends MessageKind("Declaration")
+  case NotFound extends MessageKind("Not Found")
+  case PatternMatch extends MessageKind("Pattern Match")
+  case Cyclic extends MessageKind("Cyclic")
+  case Reference extends MessageKind("Reference")
+  case DocComment extends MessageKind("Doc Comment")
+  case LossyConversion extends MessageKind("Lossy Conversion")
+  case PatternMatchExhaustivity extends MessageKind("Pattern Match Exhaustivity")
+  case MatchCaseUnreachable extends MessageKind("Match Case Unreachable")
+  case Compatibility extends MessageKind("Compatibility")
+  case PotentialIssue extends MessageKind("Potential Issue")
+
+  override def toString: String = name
```

Comment:
Might just be my brain isn't working this morning, but I wanted a way to not have to define `name` here and instead just do:

```scala
enum Foo:
  case One
  case TwoWord
```
And then just provide a `toString` that would change `TwoWord` into `Two Word` but leave `One` as is, however I was having a hard time figuring out how to actually access the `$new.$name` as a fallback. Is there an easy cheap way to achieve this that I may be missing to avoid having to pass in `name` here?

Reply:
Actually maybe nevermind, since not all of them follow that pattern. ex: `Match case Unreachable`.